### PR TITLE
fix (\) SparseVector in SuiteSparse.CHOLMOD

### PR DESCRIPTION
**Pull request by pochoi**
(\\) for Factor and SparseVecOrMat has the following method error:
```julia
julia> using SparseArrays, LinearAlgebra
julia> sparseI = sparse(1.0I, 100, 100);
julia> sparseb = sprandn(100, 0.5); #SparseVector
julia> chI = cholesky(sparseI); #Factor
julia> chI \ sparseb ≈ sparseb #(\)(Factor, SparseVecOrMat)
ERROR: MethodError: no method matching SuiteSparse.CHOLMOD.Sparse(::SparseVector{Float64,Int64}, ::Int64)
Closest candidates are:
  SuiteSparse.CHOLMOD.Sparse(::Integer, ::Integer, ::Array{Int64,1}, ::Array{Int64,1}, ::Array{#s608,1} where #s608<:Union{Complex{Float64}, Float64}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/SuiteSparse/src/cholmod.jl:882
  SuiteSparse.CHOLMOD.Sparse(::Integer, ::Integer, ::Array{Int64,1}, ::Array{Int64,1}, ::Array{Tv<:Union{Complex{Float64}, Float64},1}, ::Any) where Tv<:Union{Complex{Float64}, Float64} at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/SuiteSparse/src/cholmod.jl:848
  SuiteSparse.CHOLMOD.Sparse(::SparseMatrixCSC{Tv<:Union{Complex{Float64}, Float64},Int64}, ::Integer) where Tv<:Union{Complex{Float64}, Float64} at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/SuiteSparse/src/cholmod.jl:896
  ...
Stacktrace:
 [1] \(::SuiteSparse.CHOLMOD.Factor{Float64}, ::SparseVector{Float64,Int64}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/SuiteSparse/src/cholmod.jl:1718
 [2] top-level scope at none:0
```

The method error is due to the call `Sparse(A::SparseMatrixCSC{Tv,SuiteSparse_long}, stype::Integer)`. 
It should call `Sparse(A::SparseVector{<:VTypes,SuiteSparse_long})` and `Sparse(A::SparseMatrixCSC{<:VTypes,<:ITypes})` instead.

I fixed it and wrote a test:
```julia
Test Summary:                        | Pass  Total
Test \ for Factor and SparseVecOrMat |    2      2
```

### CODE REVIEW

File: stdlib/SuiteSparse/src/cholmod.jl [hunk 1]
```diff
@@ -1714,7 +1714,7 @@ end
 (\)(L::Factor{T}, B::StridedMatrix) where {T<:VTypes} = Matrix(L\Dense{T}(B))
 (\)(L::Factor, B::Sparse) = spsolve(CHOLMOD_A, L, B)
 # When right hand side is sparse, we have to ensure that the rhs is not marked as symmetric.
-(\)(L::Factor, B::SparseVecOrMat) = sparse(spsolve(CHOLMOD_A, L, Sparse(B, 0)))
+(\)(L::Factor, B::SparseVecOrMat) = sparse(spsolve(CHOLMOD_A, L, Sparse(B)))
```
Comment (andreasnoack) [inline stdlib/SuiteSparse/src/cholmod.jl]: It's not completely clear but the comment right above is the explanation for the zero so it is wrong to remove it here. Instead, I'd just split the method into one for matrices and one for vectors.

**Review by andreasnoack — CHANGES_REQUESTED** [review]
See inline comment.

**Review by andreasnoack — APPROVED** [review]

**Comment by pochoi** [comment]
Thanks so much for the comment. I also wrote a test for the symmetric matrix case.

**Comment by ViralBShah** [comment]
@pochoi Let's get this merged. I've submitted a new PR to rebase this PR on master and see if CI is good with it.

If you are able to rebase this PR to master I will close #34989.

**Comment by dkarrasch** [comment]
In case this gets successfully rebased on master, I'd suggest https://github.com/JuliaLang/julia/pull/34989#pullrequestreview-368545590.

**Comment by ViralBShah** [comment]
I've merged #34989

**Comment by pochoi** [comment]
Thank you so much!

**Pull request closed without merging**